crates/outbound-pg/src/lib.rs

```diff
@@ -35,27 +44,43 @@ impl HostComponent for OutboundPg {
 }
 
 #[async_trait]
-impl postgres::Host for OutboundPg {
+impl v2::Host for OutboundPg {}
+
+#[async_trait]
+impl v2::HostConnection for OutboundPg {
+    async fn open(&mut self, address: String) -> Result<Result<Resource<Connection>, v2::Error>> {
+        Ok(async {
+            self.connections
+                .push(
+                    build_client(&address)
+                        .await
+                        .map_err(|e| v2::Error::ConnectionFailed(format!("{:?}", e)))?,
+                )
+                .map(Resource::new_own)
+                .map_err(|_| v2::Error::OtherError("too many connections".into()))
```

Comment:
I know it was just a reading failure, but it took me a minute to realize that the `map_err` wasn't applying to `Resource::new_own`

```suggestion
                .map_err(|_| v2::Error::OtherError("too many connections".into()))
                .map(Resource::new_own)
```